Sources/NIOPosix/Bootstrap.swift

```diff
@@ -482,6 +482,484 @@ public final class ServerBootstrap {
+                    try serverChannel.pipeline.syncOperations.addHandler(
+                        AcceptHandler(childChannelInitializer: childChannelInit, childChannelOptions: childChannelOptions),
+                        name: "AcceptHandler"
+                    )
+
+                    // We are wrapping the inbound channels into `NIOAsyncChannel`s with the transformation
+                    // closure of the `NIOAsyncChannel` that allows us to wrap them without adding
+                    // wrapping/unwrapping handlers to the pipeline.
+                    let asyncChannel = try NIOAsyncChannel<NIOAsyncChannel<ChildChannelInbound, ChildChannelOutbound>, Never>(
+                        synchronouslyWrapping: serverChannel,
+                        backpressureStrategy: serverBackpressureStrategy,
+                        transformationClosure: { channel in
+                            try NIOAsyncChannel(
+                                synchronouslyWrapping: channel,
+                                backpressureStrategy: childBackpressureStrategy,
+                                isOutboundHalfClosureEnabled: isChildChannelOutboundHalfClosureEnabled,
+                                inboundType: ChildChannelInbound.self,
+                                outboundType: ChildChannelOutbound.self
+                            )
+                        }
+
+                    )
+                    return register(eventLoop, serverChannel).map { asyncChannel }
+                } catch {
+                    return eventLoop.makeFailedFuture(error)
+                }
+            }.flatMapError { error in
+                serverChannel.close0(error: error, mode: .all, promise: nil)
+                return eventLoop.makeFailedFuture(error)
+            }
+        }.flatMap {
+            $0
+        }.get()
+    }
+}
+
+// MARK: AsyncChannel based bind with protocol negotiation
+extension ServerBootstrap {
+    /// Bind the `ServerSocketChannel` to the `host` and `port` parameters.
+    /// Waiting for protocol negotiation to finish before yielding the channel to the returned ``NIOAsyncChannel``.
+    ///
+    /// - Parameters:
+    ///   - host: The host to bind on.
+    ///   - port: The port to bind on.
+    ///   - protocolNegotiationHandlerType: The protocol negotiation handler type that is awaited on. A handler of this type
+    ///   must be added to the pipeline in the child channel initializer.
+    ///   - serverBackpressureStrategy: The back pressure strategy used by the server socket channel.
+    /// - Returns: A ``NIOAsyncChannel`` of  the protocol negotiation results. It is expected that the protocol negotiation handler
+    /// is going to wrap the child channels into ``NIOAsyncChannel`` which are returned as part of the negotiation result.
+    @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
+    @_spi(AsyncChannel)
+    public func bindWithProtocolNegotiation<Handler: NIOProtocolNegotiationHandler>(
```

Comment:
Your other `bind`s for protocol negotiation don't have the `WithProtocolNegotiation` -- I think dropping it is fine since you're require to specify the `protocolNegotiationHandlerType`.

Reply:
Thanks for catching this! This was necessary at some point when I had no way to disambiguate them.